force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -252,36 +301,53 @@ public with sharing class ClassesObjectsMethods {
      */
     public static String createAndGetBookDetails() {
         // Create a Book object with title "A Brief History of Time" and author "Stephen Hawking"
+        String expectedTitle = 'A Brief History of Time';
+        String expectedAuthor = 'Stephen Hawking';
+        ClassesObjectsMethods.Book myBook = new ClassesObjectsMethods.Book(expectedTitle, expectedAuthor);
 
         // Call the getBookDetails method on the book object to get the details of the book
+        String actualDetails = myBook.getBookDetails();
 
         // Return the details of the book
-        return null; // Replace null with the variable you used to store the result
+        return actualDetails;
     }
 
     /**
      * Question 13
      * Create and update two Book objects and returns the books.
-     * For example, generateBookList(Book) should return a list like ["Title: Harry Potter and the Chamber of Secrets, Author: J.K. Rowling", "Title: A Brief History of Time, Author: Stephen Hawking"].
+     * For example, generateBookList(Book) should return a list like ["Title: Harry Potter and the Chamber of Secrets, Author: J.K. Rowling", 
+     *          "Title: A Brief History of Time, Author: Stephen Hawking"].
      * Update Book 1 values: title = "Harry Potter and the Chamber of Secrets", author = "J.K. Rowling", pages = 352
      * Book 2 values: title = "A Brief History of Time", author = "Stephen Hawking"
      * @param myBook1 The first book object.
      * @return return a list of books.
      */
     public static List<Book> generateBookList(Book myBook1) {
+        String expectedTitle = 'A Brief History of Time';
+        String expectedAuthor = 'Stephen Hawking';
         // Create a list of books
+        List<Book> bookList = new List<Book>();
 
-        // Update the title and author of myBook1
+        // Create a Book object with title "Harry Potter and the Chamber of Secrets" and author "J.K. Rowling"
+        ClassesObjectsMethods.Book myBookNew = new ClassesObjectsMethods.Book(expectedTitle, expectedAuthor);
 
-        // Update the pages for myBook1 to 352
+        // Update the title and author of myBook1
+        myBookNew.title = 'Harry Potter and the Chamber of Secrets';
+        myBookNew.author = 'J.K. Rowling';
+        // Update the pages for myBookNew to 352
+        myBookNew.pages = 352;
 
         // Create a new Book object with title "A Brief History of Time" and author "Stephen Hawking"
+        ClassesObjectsMethods.Book myBook2 = new ClassesObjectsMethods.Book('A Brief History of Time', 'Stephen Hawking');
 
         // Update the pages for A Brief History of Time to 256
+        myBook2.pages = 256;
 
         // Add the book to the list
-        
-        return null; // Replace null with the variable you used to store the result      
+        booklist.add(myBookNew);
+        booklist.add(myBook2);
+
+        return bookList; 
```

Comment:
Confusing code. You create a new Book object. myBookNew rather than just updating the values in the myBook1 object. Then overwrite the values in myBookNew then create a replacement for myBook1 with myBoo2.

IT works but it is more code than you need and hard to follow the logic